src/gt4py/next/iterator/transforms/constant_folding.py

```diff
@@ -20,38 +20,129 @@ def visit_FunCall(self, node: ir.FunCall):
         new_node = self.generic_visit(node)
 
-        if (
-            isinstance(new_node.fun, ir.SymRef)
-            and new_node.fun.id in ["minimum", "maximum"]
-            and new_node.args[0] == new_node.args[1]
-        ):  # `minimum(a, a)` -> `a`
-            return new_node.args[0]
-
-        if (
-            isinstance(new_node.fun, ir.SymRef)
-            and new_node.fun.id == "if_"
-            and isinstance(new_node.args[0], ir.Literal)
-        ):  # `if_(True, true_branch, false_branch)` -> `true_branch`
+        if cpm.is_call_to(new_node, ("minimum", "maximum")):
+            # `minimum(a, a)` -> `a`
+            if new_node.args[0] == new_node.args[1]:
+                new_node = new_node.args[0]
+        # `maximum(maximum(__out_size_1, 1), maximum(1, __out_size_1))` -> `maximum(__out_size_1, 1)`
+        if cpm.is_call_to(new_node, ("minimum", "maximum")):
+            if all(cpm.is_call_to(arg, "maximum") for arg in new_node.args) or all(
+                cpm.is_call_to(arg, "minimum") for arg in new_node.args
+            ):
+                if (
+                    new_node.args[0].args[0] == new_node.args[1].args[1]
+                    and new_node.args[0].args[1] == new_node.args[1].args[0]
+                ):
+                    new_node = new_node.args[0]
+        # `maximum(maximum(__out_size_1, 1), __out_size_1)` -> `maximum(__out_size_1, 1)`
+        if cpm.is_call_to(new_node, ("minimum", "maximum")):
+            match = False
+            if isinstance(new_node.args[0], ir.FunCall) and isinstance(
+                new_node.args[1], (ir.Literal, ir.SymRef)
+            ):
+                fun_call, sym_lit = new_node.args
+                match = True
+            elif isinstance(new_node.args[0], (ir.Literal, ir.SymRef)) and isinstance(
+                new_node.args[1], ir.FunCall
+            ):
+                match = True
+                sym_lit, fun_call = new_node.args
+            if match and cpm.is_call_to(fun_call, ("maximum", "minimum")):
+                if isinstance(fun_call.args[0], ir.SymRef) and isinstance(
+                    fun_call.args[1], ir.Literal
+                ):
+                    if sym_lit == fun_call.args[0]:
+                        new_node = im.call(fun_call.fun.id)(sym_lit, fun_call.args[1])
+                    if sym_lit == fun_call.args[1]:
+                        new_node = im.call(fun_call.fun.id)(fun_call.args[0], sym_lit)
+                if isinstance(fun_call.args[0], ir.Literal) and isinstance(
+                    fun_call.args[1], ir.SymRef
+                ):
+                    if sym_lit == fun_call.args[0]:
+                        new_node = im.call(fun_call.fun.id)(fun_call.args[1], sym_lit)
+                    if sym_lit == fun_call.args[1]:
+                        new_node = im.call(fun_call.fun.id)(sym_lit, fun_call.args[0])
+        # `maximum(plus(__out_size_1, 1), minus(__out_size_1, 1))` -> `plus(__out_size_1, 1)`
```

Comment:
```suggestion
        # `maximum(plus(__out_size_1, 1), minus(__out_size_1, 1))` -> `plus(__out_size_1, 1)`
```
Let's solve this by a canonicalization
`minus(literal) -> plus(-literal)`
followed by a simplification
`maximum(plus(a, literal1), plus(a, literal2))` -> `plus(a, maximum(literal1, literal2))` -> plus(a, constant_folded_literal)